fitbenchmarking/resproc/visual_pages.py

```diff
@@ -52,133 +51,76 @@ def create(prob_results, group_name, results_dir, count):
-        create_rst_page(best_result.problem.name, fig_data, fig_start, fig_fit,
-                        best_result.minimizer, see_also_link, ini_details_tbl,
-                        fin_details_tbl)
-    save_page(rst_text, problem_name, file_path)
-
-    return rst_link
-
-
-def create_rst_page(name, fig_data, fig_start, fig_fit, minimizer,
-                    see_also_link, ini_det_tbl, fin_det_tbl):
-    """
-    Creates an rst page containing a title and 3 figures, with detailed
-    tables about the fit on the last two figures.
-
-    @param name :: string containing the name of the problem
-    @param figure_data :: path to the raw plot of the data (.png image)
-    @param figure_start :: path to the starting guess plot (.png image)
-    @param figure_fit :: path to the best fit plot (.png image)
-    @param details_table :: the plot details table in rst
-    @param minimizer :: name of the minimizer that was used
-    @param see_also_link :: links to an alternative page
-                            if documentation on the problem exists.
-
-    @returns :: the visual display page written in rst
-    """
-
-    space = "|\n|\n|\n\n"
-    title = generate_rst_title(name)
-    data_plot = generate_rst_data_plot(fig_data)
-    starting_plot = generate_rst_starting_plot(fig_start, ini_det_tbl)
-    solution_plot = generate_rst_solution_plot(fig_fit, minimizer, fin_det_tbl)
-
-    rst_text = title + space + data_plot + starting_plot + solution_plot + \
-        space + see_also_link
-
-    return rst_text
-
-
-def process_problem_name(problem_name):
-    """
-    Remove commas and replace space with underscore in problem name
-
-    @param problem_name :: name of the problem that is being considered
-
-    @returns :: the processed problem name
-    """
-
-    problem_name = problem_name.replace(',', '')
-    problem_name = problem_name.replace(' ', '_')
-
-    return problem_name
+        get_figure_paths(support_pages_dir, prob_name, count)
 
+    root = os.path.dirname(os.path.abspath(__file__))
+    env = Environment(loader=FileSystemLoader(root))
 
-def generate_rst_link(file_path):
-    """
-    Generate the rst link, dependent on the OS, i.e. windows needs
-    an extra file:/// at the start to link correctly.
-
-    @param file_path :: the path to the visual display page
-
-    @returns :: the proper rst_link
-    """
-
-    rst_file_path = file_path.replace('\\', '/')
-    if os.name == 'nt':
-        rst_link = "<file:///" + rst_file_path + "." + FILENAME_EXT_HTML + \
-                   ">`__"
-    else:
-        rst_link = "<" + rst_file_path + "." + FILENAME_EXT_HTML + ">`__"
+    template = env.get_template('results_template.html')
+    html_link = "{0}.html".format(file_path)
 
-    return rst_link
+    with open(html_link, 'w') as fh:
+        fh.write(template.render(
+            title=prob_name,
+            equation=best_result.problem.equation,
+            initial_guess=best_result.ini_function_def.split("|")[1],
```

Comment:
This "eq | params" format was exclusively used by the old version of this file... we should change where this is saved to the result rather than splitting it here. Same for the "min_params"

Reply:
done